packages/modal/src/ModalView.tsx

```diff
@@ -0,0 +1,300 @@
+import React, { useCallback, useState } from 'react';
```

Comment:
This code was originally in `Modal`, I moved it out since I wrapped `Modal` in the `<PopoverProvider>` and I wanted to consume the provider values.